### PR TITLE
frontend: Fix recent search item eviction

### LINKED ISSUE #6778: Global search evicts a recent item when selecting an existing entry
## Description

Selecting an existing Global Search recent item while the recent-item list is at capacity incorrectly removes the oldest entry.

Because the selected item already exists in the list, selecting it again should only update its timestamp and move it to the top. It should not remove another recent item.

## Steps to Reproduce

1. Start Headlamp and connect it to a Kubernetes cluster.
2. Open Global Search using the search bar or the `/` keyboard shortcut.
3. Select 10 distinct search results to fill the recent-item list.
4. Open Global Search again with an empty query.
5. Select an existing recent item that is not the oldest entry.
6. Inspect the `search-recent-items` value in browser local storage.

The number of stored recent items can be checked from the browser console:

```js
Object.keys(
  JSON.parse(localStorage.getItem('search-recent-items') || '{}')
).length;
```

## Current Behavior

After selecting an existing recent item, the recent-item list incorrectly shrinks from 10 entries to 9 entries.

The current eviction condition assumes that every selection adds a new item:

```ts
if (entries.length + 1 > maxItems) {
  // Remove the oldest entry.
}
```

When the list is already at capacity, this condition is true even if the selected ID already exists. As a result, the oldest item is removed unnecessarily.

Additionally, when `maxItems` is `0`, the hook attempts to access an oldest entry from an empty collection and throws an error.

## Expected Behavior

- Selecting an existing recent item should update its timestamp.
- Updating an existing item should not remove any other recent item.
- Adding a new item at capacity should remove only the oldest entry.
- Setting `maxItems` to `0` should safely keep the recent-item list empty.

## Affected Code

The issue is located in:

```text
frontend/src/components/globalSearch/useRecent.tsx
```

The hook is used by Global Search with the `search-recent-items` local-storage key.

## Proposed Fix

Determine whether the selected ID is new before evicting an existing entry:

```ts
const isNewItem = !(id in recent);

if (isNewItem && entries.length >= maxItems) {
  // Remove the oldest entry.
}
```

The hook should also return an empty collection when `maxItems` is less than or equal to zero.

## Testing

Regression tests should cover the following cases:

- Selecting an existing item while the list is at capacity does not evict another item.
- Adding a new item while the list is at capacity evicts the oldest item.
- A zero-item limit keeps the list empty without throwing an error.

## Environment

- Headlamp: local development build from `main`
- Kubernetes cluster: Minikube
- Operating system: Windows
- Browser: Chromium-based browser

### PR DESCRIPTION
**Pull request by hazelmayank**
## Summary

This PR fixes incorrect eviction from the Global Search recent-item list.

When the list was at capacity, selecting an item that was already present removed the oldest entry. Selecting an existing item should only refresh its timestamp without changing the number of stored items.

The change also prevents an error when `maxItems` is set to zero.

## Related Issue

Fixes #6778

## Changes

- Check whether a selected item is new before evicting an entry.
- Evict the oldest entry only when adding a new item at capacity.
- Preserve all existing entries when refreshing a recent item.
- Safely keep the collection empty when `maxItems` is zero.
- Add unit tests covering updates, eviction, and zero capacity.

## Steps to Test

1. Start the Headlamp frontend and backend.
2. Open Global Search using the search bar or the `/` keyboard shortcut.
3. Select 10 distinct results to fill the recent-item list.
4. Open Global Search again with an empty query.
5. Select an existing recent item.
6. Confirm that the item is moved to the top without removing another entry.

The stored item count can also be verified in the browser console:

```js
Object.keys(
  JSON.parse(localStorage.getItem('search-recent-items') || '{}')
).length;
```

The count should remain unchanged after selecting an existing recent item.

Automated tests were run with:

```bash
cd frontend
npm test -- --run src/components/globalSearch/useRecent.test.tsx
npm test -- --run
```

Validation results:

- Focused tests: 3 passed
- Full frontend suite: 183 test files and 2398 tests passed
- TypeScript check passed
- Changed files passed ESLint and Prettier checks

## Screenshots (if applicable)

Not applicable. This change fixes state-management behavior and does not alter the visual layout.

## Notes for the Reviewer

The change is scoped to the recent-item eviction logic. It does not change any public APIs or add dependencies.

### CODE REVIEW

**Review by illume — CHANGES_REQUESTED** [review]
Thanks for these changes.


Can you please address the open review comments? Once you've resolved each one, please mark it as resolved.

File: frontend/src/components/globalSearch/useRecent.tsx [hunk 1]
```diff
@@ -35,23 +35,25 @@ export function useRecent(
 
   const bump = useCallback((id: string) => {
     setRecent(recent => {
+      if (maxItems <= 0) {
+        return {};
+      }
+
       const entries = Object.entries(recent);
       const newRecent: Record<string, number> = { ...recent };
+      const isNewItem = !(id in recent);
```
[the remark this answers is not in the document]
Comment (hazelmayank) [reply frontend/src/components/globalSearch/useRecent.tsx]: Addressed: replaced the in operator with an own-property check.

**Comment by hazelmayank** [comment]
> Thanks for these changes.
> 
> Can you please address the open review comments? Once you've resolved each one, please mark it as resolved.

Done with these changes @illume
